processing/src/main/java/io/druid/query/timeseries/TimeseriesQueryQueryToolChest.java

```diff
@@ -239,7 +240,8 @@ public Object apply(final Result<TimeseriesResultValue> input)
       TimeseriesQuery query, MetricManipulationFn fn
   )
   {
-    return makeComputeManipulatorFn(query, fn, true);
+    boolean shouldFinalize = MetricManipulatorFns.finalizing().equals(fn);
+    return makeComputeManipulatorFn(query, fn, shouldFinalize);
```

Comment:
The equals check is brittle, I think it'd be better to do a check against a query context key.

Or alternatively, do something like groupBy does, and use `mergeResults` to modify the query to remove the post-aggregators. Then, everything downstream of mergeResults on the outermost node (typically the broker) will ignore the post-aggregators. That could be done by overriding `doRun` in the ResultMergeQueryRunner.